src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -187,5 +187,45 @@ public static bool StartsWith<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> valu
             int valueLength = value.Length;
             return valueLength <= span.Length && SpanHelpers.SequenceEqual(ref span.DangerousGetPinnableReference(), ref value.DangerousGetPinnableReference(), valueLength);
         }
+
+        /// <summary>
+        /// Determines whether the specified sequence appears at the start of the span.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ReadOnlySpan<char> AsSpan(this string text)
+        {
+            if (text == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.text);
+
+            int textLength = text.Length;
+
+            if (textLength == 0) return ReadOnlySpan<char>.Empty;
+
+            unsafe
+            {
+                fixed (char* charPointer = text)
```

Comment:
> If you want to add the implementation to corefx, it should be the proper performant implementation.

Of course. Once the PR is unblocked after an API review, I will fix the implementation based on what you suggested:
- Make this internal constructor public (but do not add it to contract)
- Add a public method to CoreLib somewhere that returns byref to first string character (again, this method would not be in the contract – at least for now). However, it would be goodness to have this method public because of it is a key building block to allow experiments with non-UTF16 internal System.String representations. 
- Have different implementations for slow span and fast in CoreLib for the string Slice method.

I will make these changes after the API review if we want these APIs.